#
# This patch removes the in-line version of llseek.  This is required to
# get dosfstools to compile on later versions of glibc which have NPTL
# errno macro.
# The patch is modeled upon the one found in the dosfstools-2.8 source
# rpm from redhat.
#
--- mkdosfs/mkdosfs.c~	2000-11-28 09:23:58.000000000 -0700
+++ mkdosfs/mkdosfs.c	2003-08-22 16:31:50.000000000 -0600
@@ -92,31 +92,6 @@
 
 #endif /* __BIG_ENDIAN */
 
-/* Use the _llseek system call directly, because there (once?) was a bug in
- * the glibc implementation of it. */
-#include <linux/unistd.h>
-#if defined(__alpha) || defined(__ia64__)
-/* On alpha, the syscall is simply lseek, because it's a 64 bit system. */
-static loff_t llseek( int fd, loff_t offset, int whence )
-{
-    return lseek(fd, offset, whence);
-}
-#else
-# ifndef __NR__llseek
-# error _llseek system call not present
-# endif
-static _syscall5( int, _llseek, uint, fd, ulong, hi, ulong, lo,
-		  loff_t *, res, uint, wh );
-static loff_t llseek( int fd, loff_t offset, int whence )
-{
-    loff_t actual;
-
-    if (_llseek(fd, offset>>32, offset&0xffffffff, &actual, whence) != 0)
-	return (loff_t)-1;
-    return actual;
-}
-#endif
-
 /* Constant definitions */
 
 #define TRUE 1			/* Boolean constants */
--- dosfsck/io.c~	2000-11-28 09:22:42.000000000 -0700
+++ dosfsck/io.c	2003-08-22 16:32:15.000000000 -0600
@@ -39,33 +39,6 @@
 
 unsigned device_no;
 
-/* Use the _llseek system call directly, because there (once?) was a bug in
- * the glibc implementation of it. */
-#include <linux/unistd.h>
-#if defined(__alpha) || defined (__ia64__)
-/* On alpha, the syscall is simply lseek, because it's a 64 bit system. */
-static loff_t llseek( int fd, loff_t offset, int whence )
-{
-    return lseek(fd, offset, whence);
-}
-#else
-# ifndef __NR__llseek
-# error _llseek system call not present
-# endif
-static _syscall5( int, _llseek, uint, fd, ulong, hi, ulong, lo,
-		  loff_t *, res, uint, wh );
-
-static loff_t llseek( int fd, loff_t offset, int whence )
-{
-    loff_t actual;
-
-    if (_llseek(fd, offset>>32, offset&0xffffffff, &actual, whence) != 0)
-	return (loff_t)-1;
-    return actual;
-}
-#endif
-
-
 void fs_open(char *path,int rw)
 {
     struct stat stbuf;
